Dockerfile

```diff
@@ -1,22 +1,62 @@
-# Set the python version to use. This may be overriden during the build process by passing
-# --build-arg PY_VERSION=<desired version>
-# Note that the desired version must be of pattern <major>.<minor> .
-ARG PY_VERSION=3.7
-FROM python:$PY_VERSION AS cache
+FROM python:3.7-slim AS compile-image
 
-RUN pip install raiden
+LABEL maintainer=devops@brainbot.com
 
-FROM python:${PY_VERSION}
+################################################################################
+# COMPILATION STAGE (compile-image)                                            #
+################################################################################
+#
+# Install the raiden client and scenario player from their github repositoires.
+#
+# This stage supports the following, optional build args:
+#
+#   - RAIDEN_VERSION (default: develop)
+#     The version of the Raiden client to run the nightlies against.
+#
+#   - RAIDEN_VERSION (default: dev)
+#     The version of the SP to run the nightlies with.
+#
+# The build stage uses a virtual environment to install the raiden client
+# and scenario player tool, which can be copied to subsequent build stages,
+# in order to reduce image size.
 
-ARG PY_VERSION
+ARG RAIDEN_VERSION=develop
+ARG SP_VERSION=dev
 
-# Copy raiden repository and site-packages from build cache
-COPY --from=cache /usr/local/lib/python${PY_VERSION}/dist-packages /usr/local/lib/python${PY_VERSION}/dist-packages
+# Set up the build stage
+RUN apt-get update
+RUN apt-get install -y --no-install-recommends build-essential gcc git
+RUN python -m venv /opt/venv
+ENV PATH="/opt/venv/bin:$PATH"
 
-#  Copy SP folder and install.
-CMD mkdir /scenario-player
-ADD . /scenario-player
+# Install the Scenario Player
+RUN git clone https://github.com/raiden-network/scenario-player.git /sp
+WORKDIR /sp
+RUN git checkout ${SP_VERSION}
+RUN pip install .
 
-RUN pip install ./scenario-player
+# Install the Raiden Client
+RUN git clone https://github.com/raiden-network/raiden.git /raiden
+WORKDIR /raiden
+RUN git checkout ${RAIDEN_VERSION}
+RUN pip install .
 
-ENTRYPOINT ["/usr/local/bin/scenario_player"]
+FROM python:3.7-slim as execution-image
+WORKDIR /
+
+################################################################################
+# PREPARATION STEP (execution-image)                                           #
+################################################################################
+#
+# Copies the raiden client, scenario player and scenario defnition files from the
+# compilation stage, and configure further environment variables required to run
+# the nightlies.
+#
+# This stage copies the virtual environment created in the previous stage,
+# and prepends its location to the $PATH environment variable.
+
+# Copy virtual env and scenario definition files.
+COPY --from=compile-image /opt/venv /opt/venv
```

Comment:
Depending on how you intend to use this, it would make sense to copy the scenarios from the raiden checkout above.

Reply:
The alternative is of course to mount the scenario descriptions into the running container (which will need another git checkout step previous to the execution)

Reply:
Scenarios are now part of the container